File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

    HashMap<Integer, Integer> freqq = new HashMap<>();

        for (int i =0;i<nums.length;i++){
            if(freqq.containsKey(nums[i])){
              freqq.put(nums[i],freqq.get(nums[i]) +1);
            }else{
             freqq.put(nums[i] , 1);
            }
        }
   

       PriorityQueue<Integer> heap = new PriorityQueue<>((a, b) -> freqq.get(b) - freqq.get(a));
        for (int num : freqq.keySet()) {
            heap.offer(num);
        }
        
        int[] result = new int[k];
        for (int i = 0; i < k; i++) {
            result[i] = heap.poll();
        }
      return result;
    }
}

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        frequent={}
        for num in nums:
            if num in frequent:
                frequent[num]+=1
            else:
                frequent[num]=1
        sorted_values=[key for key, _ in(sorted(frequent.items(), key=lambda item: item[1], reverse=True))]
        return(sorted_values[:k])
        # print(sorted_values[:2].keys()) 
        